### PR TITLE
[libc++] LWG3382: NTTP for `pair` and `array`

### PR DESCRIPTION
**Pull request by JMazurkiewicz**
Mark LWG3382 as "Nothing To Do" and add tests.

### CODE REVIEW

File: libcxx/test/std/containers/sequences/array/array.overview/nttp.equivalence.compile.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+
+// <array>
+
+// Two values a1 and a2 of type array<T, N> are template-argument-equivalent if and only if each pair of corresponding
```
Comment (ldionne) [inline libcxx/test/std/containers/sequences/array/array.overview/nttp.equivalence.compile.pass.cpp:13]: Can you add references to the LWG issue being tested here? And also in the one for `pair`.

File: libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+
+// <array>
+
+// array<T, N> is a structural type ([temp.param]) if T is a structural type.
```
Comment (ldionne) [inline libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp:13]: I don't understand what you're testing here. Isn't it sufficient to check for the NTTP behavior as far as the LWG issue is concerned? I think I'm just missing the obvious here, but if you can explain that would be great!

**Review by ldionne — COMMENTED** [review]
Thanks for tackling this! I have some comments to clarify my understanding but I think this will be good without too many changes.

File: libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp (hunk 2 above)
Comment (JMazurkiewicz) [reply libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp:13]: I'm testing whether `array` (or `pair`) is a structural type based on its template parameters. LWG-3382 states that:
* pair<T, U> is a structural type (13.2 [[temp.param]](https://wg21.link/temp.param)) if T and U are both structural types
* array<T, N> is a structural type (13.2 [[temp.param]](https://wg21.link/temp.param)) if T is a structural type

IMHO those tests address the LWG issue, although they might be an overkill.

**Review by ldionne — APPROVED** [review]

**Pull request merged**